gadgets/trace_mount/program.bpf.c

```diff
@@ -55,20 +56,9 @@ struct {
 	__type(value, struct arg);
 } args SEC(".maps");
 
-struct {
-	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
-	__uint(max_entries, 1);
-	__type(key, int);
-	__type(value, struct event);
-} heap SEC(".maps");
-
-struct {
-	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
-	__uint(key_size, sizeof(__u32));
-	__uint(value_size, sizeof(__u32));
-} events SEC(".maps");
+GADGET_TRACER_MAP(events, 1024 * 256);
```

Comment:
Perhaps add a comment indicating how many events `1024 * 256` corresponds to?

Reply:
What's the issue with `sizeof(struct event)`? Is it that the size has to be a power of 2? Or is there another issue?

Reply:
Right, it's because the sizeof(struct event) * N is not a power of two. The following makes it work but wastes a lot of space:


```diff
diff --git gadgets/trace_mount/program.bpf.c gadgets/trace_mount/program.bpf.c
index 6be76879..48210e7e 100644
--- gadgets/trace_mount/program.bpf.c
+++ gadgets/trace_mount/program.bpf.c
@@ -43,6 +43,8 @@ struct event {
        __u8 dest[PATH_MAX];
        __u8 data[DATA_LEN];
        enum op op;
+
+       __u8 pad[7608];
 };
 
 const volatile pid_t target_pid = 0;
@@ -57,7 +59,7 @@ struct {
 } args SEC(".maps");
 
 // Roughly 30 events
-GADGET_TRACER_MAP(events, 1024 * 256);
+GADGET_TRACER_MAP(events, sizeof(struct event)*32);
 
 GADGET_TRACER(mount, events, event);
```


 it'd be nice to have a macro to round to the next power of 2, but I couldn't find it -> https://stackoverflow.com/questions/22925016/rounding-up-to-powers-of-2-with-preprocessor-constants.